ratis-common/src/main/java/org/apache/ratis/util/Daemon.java

```diff
@@ -49,6 +60,11 @@ public Builder setRunnable(Runnable runnable) {
       return this;
     }
 
+    public Builder setUncaughtExceptionHandler(UncaughtExceptionHandler exceptionHandler) {
```

Comment:
Not sure how much UT would be necessary. If necessary, I'll try writing something similar to RaftStateMachineExceptionTests using a MiniRaftCluster which creates a RaftServer with my user defined UncaughtExceptionHandler, and when I throw an exception internally, check if the handler is able to catch that.